### PR TITLE
2047 taxlot dedicated map

### PR DESCRIPTION
**Pull request by lainsworth8801**
#### Any background context you want to provide?
san-jose-test-taxlot.csv as test file for validation.

#### What's this PR do?
Added taxlot map for dedicated taxlot files.

#### How should this be manually tested?
Import san-jose-test-taxlot.csv and map taxlots properties. Expect to see a dedicated map for taxlot under inventory.

#### What are the relevant tickets?
#2047 Provide dedicated Tax Lot Map page

#### Screenshots (if appropriate)
Import 6 taxlots:
Screen Shot 2020-07-24 at 7 41 56 AM

Map shows 6 taxlots:
Screen Shot 2020-07-24 at 7 43 01 AM

Added centroid for taxlots and property and taxlot taps for map:
Screen Shot 2020-08-06 at 3 52 17 PM

### CODE REVIEW

File: seed/static/seed/partials/ubid_modal.html [hunk 1]
```diff
@@ -23,7 +23,7 @@ <h4 class="modal-title" id="exportModalLabel" ng-show="decode_ubid_state === 're
   <br>
 
   <!-- The summary of selected records is shown in any state. -->
-  <div class="alert alert-info" ng-if="pre_decode_ubid_not_decoded || pre_decode_ubid_not_decoded">
+  <div class="alert alert-info" ng-if="pre_decode_ubid_not_decoded || pre_decode_ubid_not_decoded"><!--TODO: pre_decode_ulid_not_decoded?-->
```
Comment (lainsworth8801) [inline seed/static/seed/partials/ubid_modal.html:26]: Need confirmation.

Comment (adrian-lara) [reply seed/static/seed/partials/ubid_modal.html:26]: What's the question being asked here?

**Comment by adrian-lara** [comment]
I updated my initial comment with the items I'm able to see now. As far as I can tell on the UI/UX, there's just those last items 2 left to address.

File: seed/static/seed/partials/ubid_modal.html (hunk 1 above)
Comment (lainsworth8801) [reply seed/static/seed/partials/ubid_modal.html:26]: Just noticed that pre_decode_ubid_not_decoded is used twice here? Should one of them be ulid instead?

File: seed/static/seed/partials/inventory_map.html [hunk 2]
```diff
@@ -98,13 +98,22 @@ <h1>{$:: (inventory_type === 'taxlots' ? 'Tax Lots' : 'Properties') | translate
...
+        <li ng-class="::{active: inventory_type==='properties'}" heading="View by Property">
+            <a ui-sref="inventory_map(::{inventory_type: 'properties'})" translate>View by Property</a>
+        </li>
+        <li ng-class="::{active: inventory_type==='taxlots'}" heading="View by Tax Lot">
+            <a ui-sref="inventory_map(::{inventory_type: 'taxlots'})" translate>View by Tax Lot</a>
+        </li>
+    </ul>
+    <div class="label-tabs">
+        <label class="pad-left-20 pad-right-20"> <input type="checkbox" name="hexbin" ng-model="layers.hexbin_layer.visible" ng-true-value=1 ng-false-value=0 ng-change="toggle_layer(layers.hexbin_layer.zIndex)" ng-disabled="::inventory_type === 'taxlots'">&nbsp;&nbsp;Hexagonal Bins</label>
+        <label class="pad-right-20"> <input type="checkbox" name="points" ng-model="layers.points_layer.visible" ng-true-value=1 ng-false-value=0 ng-change="toggle_layer(layers.points_layer.zIndex)">&nbsp;&nbsp;Building Points</label>
```
Comment (adrian-lara) [inline seed/static/seed/partials/inventory_map.html:111]: Oh also, should we change `Building Point` to `Tax Lot Point` or something for the tax lot page? What do you think?

**Comment by adrian-lara** [comment]
> Property & Tax Lot Tabs Position:
> Screen Shot 2020-08-22 at 8 33 55 AM
> I like how this looks! One possible issue with this is how close the Hex Bin & Building Point check buttons are to the Property and Tax Lot tab buttons. @axelstudios, could we get your quick opinion on this?
> 
> This is what it looked like before the tabs:
> Screen Shot 2020-08-22 at 8 34 04 AM

@axelstudios this PR is pretty close - any strong opinions either way re ☝️ ?

File: seed/models/tax_lot_properties.py [hunk 3]
```diff
@@ -381,17 +381,15 @@ def get_related(cls, object_list, show_columns, columns_from_database):
             obj_dict[lookups['long_lat']] = long_lat_wkt(obj.state)
 
             # store the property / taxlot data to the object dictionary as well. This is hacky.
+            # bring in GIS data
+            obj_dict[lookups['centroid']] = centroid_wkt(obj.state)
```
[the remark this answers is not in the document]
Comment (lainsworth8801) [reply seed/models/tax_lot_properties.py:385]: Thanks for catching that! I was being careless and overlooked your comments.

File: seed/static/seed/partials/inventory_map.html (hunk 2 above)
Comment (lainsworth8801) [reply seed/static/seed/partials/inventory_map.html:111]: I agreed that Tax Lot Point is probably more accurate but I'd like to double check with alex/nick that "building" as a concept doesn't include "taxlot". I believe there has been a convo about the definition of "building".

**Comment by lainsworth8801** [comment]
https://github.com/SEED-platform/seed/pull/2334#discussion_r475703126

Simply bc bb_box wasn't a taxlot centroid as all taxlot properties come with dedicated centroid. I wondered if it was used as a place holder when taxlot map wasn't implemented?

File: seed/static/seed/js/controllers/inventory_map_controller.js [hunk 4]
```diff
@@ -147,8 +172,7 @@ angular.module('BE.seed.controller.inventory_map', [])
         var buildingCentroid = function (building) {
           var format = new ol.format.WKT();
 
-          var bounding_box = building.centroid;
-          var feature = format.readFeature(bounding_box, {
+          var feature = format.readFeature(building.centroid, {
```
[the remark this answers is not in the document]
Comment (lainsworth8801) [reply seed/static/seed/js/controllers/inventory_map_controller.js:152]: Simply bc bb_box wasn't a taxlot centroid as all taxlot properties come with dedicated centroid. I wondered if it was used as a place holder when taxlot map wasn't implemented?

**Comment by RDmitchell** [comment]
@adrian-lara -- I vote for more space above the map checkboxes to separate a bit more from the Property and Tax Lot tabs

File: seed/static/seed/js/controllers/inventory_map_controller.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (adrian-lara) [reply seed/static/seed/js/controllers/inventory_map_controller.js:152]: Discussed - we're good here. 👍

File: seed/static/seed/partials/inventory_map.html (hunk 2 above)
Comment (adrian-lara) [reply seed/static/seed/partials/inventory_map.html]: I see this changed to "Taxlot Points". Assuming you discussed this with others?

**Comment by adrian-lara** [comment]
2 open questions and one really small code change. Otherwise, this works as expected!

File: seed/static/seed/js/controllers/inventory_map_controller.js [hunk 5]
```diff
@@ -64,41 +60,53 @@ angular.module('BE.seed.controller.inventory_map', [])
         windowClass: 'inventory-progress-modal',
         scope: $scope
       });
-      return fetch(page, chunk).then(function (data) {
+
+      if ($scope.inventory_type == "properties") var getInventoryFunc = inventory_service.get_properties;
+      else getInventoryFunc = inventory_service.get_taxlots;
```
[the remark this answers is not in the document]
Comment (adrian-lara) [reply seed/static/seed/js/controllers/inventory_map_controller.js:66]: Even better! 👍

**Pull request merged**